siad/server.go

```diff
@@ -382,6 +383,23 @@ func NewServer(bindAddr, requiredUserAgent, requiredPassword string) (*Server, e
 		listener: l,
 		httpServer: &http.Server{
 			Handler: mux,
+
+			// set reasonable timeout windows for requests, to prevent the Sia API
+			// server from leaking file descriptors due to slow, disappearing, or
+			// unreliable API clients.
+
+			// ReadTimeout defines the maximum amount of time allowed to fully read
+			// the request body. This timeout is applied to every handler in the
+			// server.
+			ReadTimeout: time.Minute,
```

Comment:
Would like to see all of these be friendly to tunneling over Tor, though it's an uncommon use case, I see no danger in supporting it. That means 2m-5m for each value, please update